src/test/java/uk/gov/hmcts/reform/unspec/service/WorkingDayIndicatorTest.java

```diff
@@ -0,0 +1,198 @@
+import org.mockito.Mock;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import uk.gov.hmcts.reform.unspec.service.bankholidays.NonWorkingDaysCollection;
+import uk.gov.hmcts.reform.unspec.service.bankholidays.PublicHolidaysCollection;
+
+import java.time.LocalDate;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Stream;
+
+import static java.util.Collections.singletonList;
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(SpringExtension.class)
+class WorkingDayIndicatorTest {
+
+    private static final LocalDate BANK_HOLIDAY = LocalDate.of(2017, 5, 29);
+    private static final LocalDate NEXT_WORKING_DAY_AFTER_BANK_HOLIDAY = LocalDate.of(2017, 5, 30);
+    private static final LocalDate PREVIOUS_WORKING_DAY_BEFORE_BANK_HOLIDAY = LocalDate.of(2017, 5, 26);
+    private static final LocalDate SATURDAY_WEEK_BEFORE = LocalDate.of(2017, 6, 3);
+    private static final LocalDate SUNDAY_WEEK_BEFORE = LocalDate.of(2017, 6, 4);
+    private static final LocalDate MONDAY = LocalDate.of(2017, 6, 5);
+    private static final LocalDate TUESDAY = LocalDate.of(2017, 6, 6);
+    private static final LocalDate WEDNESDAY = LocalDate.of(2017, 6, 7);
+    private static final LocalDate THURSDAY = LocalDate.of(2017, 6, 8);
+    private static final LocalDate FRIDAY = LocalDate.of(2017, 6, 9);
+    private static final LocalDate SATURDAY = LocalDate.of(2017, 6, 10);
+    private static final LocalDate SUNDAY = LocalDate.of(2017, 6, 11);
+
+    static class WeekDays implements ArgumentsProvider {
+
+        @Override
+        public Stream<Arguments> provideArguments(ExtensionContext context) {
+            return Stream.of(
+                Arguments.of(MONDAY),
+                Arguments.of(TUESDAY),
+                Arguments.of(WEDNESDAY),
+                Arguments.of(THURSDAY),
+                Arguments.of(FRIDAY)
+            );
+        }
+    }
+
+    static class Weekend implements ArgumentsProvider {
+
+        @Override
+        public Stream<Arguments> provideArguments(ExtensionContext context) {
+            return Stream.of(
+                Arguments.of(SATURDAY),
+                Arguments.of(SUNDAY)
+            );
+        }
+    }
+
+    private WorkingDayIndicator service;
+
+    @Mock
+    private PublicHolidaysCollection publicHolidaysApiClient;
+    @Mock
+    private NonWorkingDaysCollection nonWorkingDaysCollection;
+
+    @BeforeEach
+    void setup() {
+        service = new WorkingDayIndicator(publicHolidaysApiClient, nonWorkingDaysCollection);
+    }
+
+    @Nested
+    class ForWeekdays {
+
+        @ParameterizedTest
+        @ArgumentsSource(WeekDays.class)
```

Comment:
could actually use enum source annotation here, probably a little less code. can then specify which days using name in combo with include / exclude

Reply:
There is no enum here, passing a list of dates that are weekdays here.